@here/olp-sdk-core/lib/utils/TileKey.ts

```diff
@@ -0,0 +1,417 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+/**
+ * Addresses a tile in a quadtree.
+ *
+ * A quadkey is defined by a row, a column, and a level. The tree has a root at level 0 that contains one
+ * single tile. On every level, each tile is divided into four children (hence, the name "quadtree").
+ *
+ * Within each [[level]], a tile is addressed with a [[row]] and [[column]]. The number
+ * of rows and columns in each level is 2 to the power of the level. This means that on level 0, only
+ * one tile exists, so there is 1 column and 1 row. On level 1, 4 tiles exist in 2 rows and 2 columns.
+ * On level 2, we have 16 tiles in 4 rows and 4 columns. And so on.
+ *
+ * @note As the JavaScript number type can hold 53 bits in its mantissa, only levels up to 26 can be
+ * represented in the number representation returned by [[`mortonCodeFromQuadKey`]].
+ */
+export interface QuadKey {
+    /** The row that contains the tile. */
+    readonly row: number;
+    /** The column that contains the tile. */
+    readonly column: number;
+    /** The level that contains the tile. */
+    readonly level: number;
+}
+
+/**
+ * The `TileKey` instances are used to address a tile in a Quad Tree.
+ *
+ * A tile key is defined by a row, a column, and a level. The tree has a root at level 0, with one
+ * single tile. On every level, each tile is divided into four children (therefore the name
+ * quadtree).
+ *
+ * Within each [[level]], any particular tile is addressed with [[row]] and [[column]]. The number
+ * of rows and columns in each level is 2 to the power of the level. This means: On level 0, only
+ * one tile exists, [[columnsAtLevel]]() and [[rowsAtLevel]]() are both 1. On level 1, 4 tiles
+ * exist, in 2 rows and 2 columns. On level 2 we have 16 tiles, in 4 rows and 4 columns. And so on.
+ *
+ * A tile key is usually created using [[fromRowColumnLevel]]() method.
+ *
+ * `TileKey` instances are immutable, all members return new instances of `TileKey` and do not
+ * modify the original object.
+ *
+ * Tile keys can be created from and converted into various alternative formats:
+ *
+ *  - [[toQuadKeyString]]() / [[fromQuadKeyString]]() - string representation 4-based
+ *  - [[toHereTile]]() / [[fromHereTile]]() - string representation 10-based
+ *  - [[toMortonCode]]() / [[fromMortonCode]]() - number representation
+ *
+ * Note - as JavaScript's number type can hold 53 bits in its mantissa, only levels up to 26 can be
+ * represented in the number representation returned by [[mortonCode]]().
+ */
+export class TileKey implements QuadKey {
+    private m_mortonCode?: number;
+    private m_hereTile?: string;
```

Comment:
Do you need to store these? Usually row/column/level should be enough.

Reply:
removed.